discretisedfield/field.py

```diff
@@ -3059,112 +3061,98 @@ def _fromovf(cls, filename):
-            datalines = np.genfromtxt(filter(lambda s: not s.startswith('#'),
-                                             lines), dtype=None)
-
-            if '1.0' in mdatalines[0]:
-                # valuedim is not in OVF1 file.
-                mdatadict['valuedim'] = datalines.shape[-1]
-
-            for line in mdatalines:
-                for mdatum in mdatalist:
-                    if mdatum in line:
-                        mdatadict[mdatum] = float(line.split()[-1])
-                        break
-
-        except UnicodeDecodeError:
-            with open(filename, 'rb') as ovffile:
-                f = ovffile.read()
-                lines = f.split(b'\n')
-
-            mdatalines = list(filter(lambda s: s.startswith(bytes('#',
-                                                                  'utf-8')),
-                                     lines))
-
-            if bytes('2.0', 'utf-8') in mdatalines[0]:
-                endian = '<'  # little-endian
-            elif bytes('1.0', 'utf-8') in mdatalines[0]:
-                endian = '>'  # big-endian
-
-            for line in mdatalines:
-                for mdatum in mdatalist:
-                    if bytes(mdatum, 'utf-8') in line:
-                        mdatadict[mdatum] = float(line.split()[-1])
-                        break
-
-            header = b'# Begin: Data Binary '
-            data_start = f.find(header)
-            header = f[data_start:(data_start + len(header) + 1)]
-
-            data_start += len(header)
-            data_end = f.find(b'# End: Data Binary ')
-
-            if b'4' in header:
-                nbytes = 4
-                formatstr = endian + 'f'
-                checkvalue = 1234567.0
-            elif b'8' in header:
-                nbytes = 8
-                formatstr = endian + 'd'
-                checkvalue = 123456789012345.0
-
-            newlines = [b'\n\r', b'\r\n', b'\n']  # ordered by length
-            for nl in newlines:
-                if f.startswith(nl, data_start):
-                    data_start += len(nl)
-                    # There is a difference between files written by OOMMF and
-                    # mumax3. OOMMF has a newline character before the "end
-                    # metadata line', whereas mumax3 does not. Therefore if the
-                    # length of the data stream is not a multiple of nbytes, we
-                    # have to subtract the length of newline character from
-                    # data_end.
-                    if (data_end - data_start) % nbytes != 0:
-                        data_end -= len(nl)
+        mdata_str = ['meshunit', 'valueunit']  # information not used
+        mdata_list = ['valuelabels', 'valueunits']  # information not used
+
+        header = {}
+        with open(filename, 'rb') as f:
+            # >>> READ HEADER <<<
+            header['ovf_version'] = next(f).decode('utf-8').split()[-1][-3]
```

Comment:
```header['ovf_version'] = int(next(f).decode('utf-8').split()[-1])``` or maybe ```*_, header['ovf_version'] = next(f).decode('utf-8').split()[-1]```. A lot of hardcoded indices are a bit difficult to read maybe.

Reply:
The file contains either `... v1.0\n"` or `"... 2.0\n"` without the `v` (!). I decide to only keep `"1"` or `"2"` (as `string`). Therefore all the hard-coded indices. So your suggestions both don't work. Maybe you have a different idea?

Reply:
@marijanbeg Is it better now?